sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -141,6 +141,37 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
         writeOptions = c.options.filterKeys(_ != "path"),
         ignoreIfExists = c.ifNotExists)
 
+    case c @ CreateTableLikeStatement(target, source, loc, ifNotExists) =>
+      def validateLocation(loc: Option[String]) = {
+        if (loc.isDefined) {
+          throw new AnalysisException("Location clause not supported for " +
+            "CREATE TABLE LIKE statement when tables are of V2 type")
```

Comment:
we can support the LOCATION clause. See `CatalogV2Utils.convertTableProperties`, we can store the location in a special table property `location`.

Reply:
@cloud-fan Thanks a lot. I was not aware of this. I will check.